src/gui/qgsattributeform.h

```diff
@@ -443,6 +443,7 @@ class GUI_EXPORT QgsAttributeForm : public QWidget
     void updateLabels();
     bool currentFormValuesFeature( QgsFeature &feature );
     bool currentFormValidConstraints( QStringList &invalidFields, QStringList &descriptions );
+    bool currentFormValidHardConstraints( QStringList &invalidFields, QStringList &descriptions );
```

Comment:
```suggestion
    bool currentFormValidHardConstraints( QStringList &invalidFields, QStringList &descriptions ) const;
```

That could be const?